File: solutions/golang/onlineshoppingservice/product.go
package onlineshopping

type Product struct {
	ID          string
	Name        string
	Description string
	Price       float64
	Quantity    int
}

func NewProduct(id, name, description string, price float64, quantity int) *Product {
	return &Product{
		ID:          id,
		Name:        name,
		Description: description,
		Price:       price,
		Quantity:    quantity,
	}
}

func (p *Product) UpdateQuantity(quantity int) {
	p.Quantity += quantity
}

func (p *Product) IsAvailable(quantity int) bool {
	return p.Quantity >= quantity
}

File: solutions/csharp/onlineshoppingservice/Product.cs
namespace OnlineShopping
{
    public class Product
    {
        public string Id { get; }
        public string Name { get; }
        public string Description { get; }
        public double Price { get; }
        private int _quantity;

        public Product(string id, string name, string description, double price, int quantity)
        {
            Id = id;
            Name = name;
            Description = description;
            Price = price;
            _quantity = quantity;
        }

        public void UpdateQuantity(int quantity)
        {
            _quantity += quantity;
        }

        public bool IsAvailable(int quantity)
        {
            return _quantity >= quantity;
        }

        public int Quantity => _quantity;
    }
}